tests/unit/s2n_random_test.c

```diff
@@ -894,6 +897,27 @@ int main(int argc, char **argv)
 {
     BEGIN_TEST_NO_INIT();
 
+    /* Feature probe */
+    {
+#if defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE) && defined(OPENSSL_NO_ENGINE)
+        FAIL_MSG("Engine feature probe failed.")
+#endif
```

Comment:
Where does the OPENSSL_NO_ENGINE define come from? Can we rely on it being defined here?

Nit: Also, this seems unnecessary to me

Reply:
`OPENSSL_NO_ENGINE` gets defined when you build openssl with the `no-engine` configuration.

> Nit: Also, this seems unnecessary to me
Why? This is specifically testing the feature probe logic is correct.

Reply:
OPENSSL_NO_ENGINE no longer part of the feature probe